File: src/main/java/com/ject/studytrip/dummy/application/dto/DummyStampInfo.java
package com.ject.studytrip.dummy.application.dto;

import com.ject.studytrip.global.util.DateUtil;
import com.ject.studytrip.stamp.domain.model.Stamp;

public record DummyStampInfo(
        Long stampId,
        String stampName,
        int stampOrder,
        String endDate,
        int totalMissions,
        int completedMissions,
        boolean completed,
        String createdAt,
        String updatedAt,
        String deletedAt) {
    public static DummyStampInfo from(Stamp stamp) {
        return new DummyStampInfo(
                stamp.getId(),
                stamp.getName(),
                stamp.getStampOrder(),
                DateUtil.formatDate(stamp.getEndDate()),
                stamp.getTotalMissions(),
                stamp.getCompletedMissions(),
                stamp.isCompleted(),
                DateUtil.formatDateTime(stamp.getCreatedAt()),
                DateUtil.formatDateTime(stamp.getUpdatedAt()),
                DateUtil.formatDateTime(stamp.getDeletedAt()));
    }
}

File: src/main/kotlin/com/ject/studytrip/dummy/application/dto/DummyStampInfo.kt
package com.ject.studytrip.dummy.application.dto

import com.ject.studytrip.global.util.DateUtil
import com.ject.studytrip.stamp.domain.model.Stamp

data class DummyStampInfo(
    val stampName: String,
    val stampOrder: Int,
    val endDate: String?,
    val totalMissions: Int,
    val completedMissions: Int,
    val completed: Boolean,
) {
    companion object {
        @JvmStatic
        fun from(stamp: Stamp): DummyStampInfo =
            DummyStampInfo(
                stamp.name,
                stamp.stampOrder,
                DateUtil.formatDate(stamp.endDate),
                stamp.totalMissions,
                stamp.completedMissions,
                stamp.isCompleted,
            )
    }
}
